### DISCUSSION TITLE
Why does `Writer.tell` return `Writer[T, Unit]` ?

### DISCUSSION THREAD

**Discussion by atais** [typelevel/cats#4236] [Q&A] [answered]
Hello.

I find it not handy at all. Let's say all my logic is based on a `Writer[List[MappingError], T]` 

and now I can (sometimes) nicely compose it with `value`:

```
val foo: Writer[List[MappingError], Foo] = Writer.value(fooValue)
```

but now with `tell`:

```
val foo: Writer[List[MappingError], Unit] = Writer.tell(List(someError))
```

how is the `Unit` ever useful? I would expect it to bind to generic `T` that composes with the rest of my writer. Or there is some way to transform it, that I dont know of :-)?
Thanks!

**Comment by armanbilge** [comment, marked as answer]
Maybe you are looking for `as`? :)
```scala
val foo: Writer[List[MappingError], Foo] = Writer.tell(List(someError)).as(fooValue)
```

More generally you can compose using `flatMap` in a for-comprehension:
```scala
for {
  foo <- Writer.value(fooValue)
  _ <- Writer.tell(List(someError))
  _ <- Writer.tell(List(anotherError))
  bar <- Writer.value(barValue)
} yield foo
```
